File: TypeScript/codewars/Typescript/Build-Tower.ts
export const towerBuilder = (nFloors: number): string[] => {
	const Space = " ";
	const Asterisk = "*";
	const output: string[] = [];
	let quantityAsterisk = 1;

	for (let index = 1; index <= nFloors; index++) {
		const blankSpace = nFloors - index;

		const line = Space.repeat(blankSpace) + Asterisk.repeat(quantityAsterisk) + Space.repeat(blankSpace);

		output.push(line);

		quantityAsterisk += 2;
	}

	return output;
};

File: Csharp/CodeWars/src/Resolutions/BuildTower.cs
namespace CodeWars.src.Resolutions;

public class BuildTower
{
    public static string[] TowerBuilder(int nFloors)
    {
        char space = ' ';
        char asterisk = '*';
        List<string> output = [];
        int quantityAsterisk = 1;

        for (int index = 1; index <= nFloors; index++)
        {
            int blankSpace = nFloors - index;

            string line = new string(space, blankSpace) +
                          new string(asterisk, quantityAsterisk) +
                          new string(space, blankSpace);

            output.Add(line);

            quantityAsterisk += 2;
        }

        return [.. output];
    }
}
